beacon-chain/rpc/validator/attester.go

```diff
@@ -136,3 +144,41 @@ func (vs *Server) ProposeAttestation(ctx context.Context, att *ethpb.Attestation
 		AttestationDataRoot: root[:],
 	}, nil
 }
+
+// waitToOneThird waits until one-third of the way through the slot
+// or the head slot equals to the input slot.
+func (vs *Server) waitToOneThird(ctx context.Context, slot uint64) {
+	_, span := trace.StartSpan(ctx, "validator.waitToOneThird")
+	defer span.End()
+
+	// Don't need to wait if head slot is already the same as requested slot.
+	if slot == vs.HeadFetcher.HeadSlot() {
+		return
+	}
+
+	// Set time out to be at start slot time + one-third of slot duration.
+	slotStartTime := slotutil.SlotStartTime(uint64(vs.GenesisTimeFetcher.GenesisTime().Unix()), slot)
+	slotOneThirdTime := slotStartTime.Unix() + int64(params.BeaconConfig().SecondsPerSlot/3)
+	waitDuration := slotOneThirdTime - time.Now().Unix()
```

Comment:
This could underflow if now is already beyond 1/3 of the slot

Reply:
That's not a concern. Any negative value will be treated like zero in time.After